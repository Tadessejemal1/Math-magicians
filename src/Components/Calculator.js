import React, { useState } from 'react';
import Button from './Button';
import Input from './Input';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  const handleButtonClick = (name) => {
    const { total: newTotal, next: newNext, operation: newOperation } = calculate(
      { total, next, operation }, name,
    );

    setTotal(newTotal);
    setNext(newNext);
    setOperation(newOperation);
  };

  return (
    <div className="calculator">
      <div className="row">
        <Input text={next || operation || total} />
      </div>
      <div className="row">
        <Button clickHandler={handleButtonClick} name="AC" />
        <Button clickHandler={handleButtonClick} name="+/-" />
        <Button clickHandler={handleButtonClick} name="%" />
        <Button clickHandler={handleButtonClick} className="btn btn-operator" name="รท" />
      </div>
      <div className="row">
        <Button clickHandler={handleButtonClick} name="7" />
        <Button clickHandler={handleButtonClick} name="8" />
        <Button clickHandler={handleButtonClick} name="9" />
        <Button clickHandler={handleButtonClick} className="btn btn-operator" name="x" />
      </div>
      <div className="row">
        <Button clickHandler={handleButtonClick} name="4" />
        <Button clickHandler={handleButtonClick} name="5" />
        <Button clickHandler={handleButtonClick} name="6" />
        <Button clickHandler={handleButtonClick} className="btn-operator" name="-" />
      </div>
      <div className="row">
        <Button clickHandler={handleButtonClick} name="1" />
        <Button clickHandler={handleButtonClick} name="2" />
        <Button clickHandler={handleButtonClick} name="3" />
        <Button clickHandler={handleButtonClick} name="+" className="btn-operator" />
      </div>
      <div className="row">
        <Button clickHandler={handleButtonClick} className="btn-zero" name="0" />
        <Button clickHandler={handleButtonClick} name="." />
        <Button clickHandler={handleButtonClick} name="=" className="btn-operator" />
      </div>
    </div>
  );
};
export default Calculator;
