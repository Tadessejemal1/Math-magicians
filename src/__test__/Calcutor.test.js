import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Calculator from '../Components/Calculator';

it('Calculator component renders correctly', () => {
  const tree = renderer.create(<Calculator />)
    .toJSON(); expect(tree).toMatchSnapshot();
});
