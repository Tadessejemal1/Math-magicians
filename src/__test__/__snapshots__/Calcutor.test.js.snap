// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<div
  className="calculator"
>
  <div
    className="row"
  >
    <div
      className="input"
    >
      0
    </div>
  </div>
  <div
    className="row"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn btn-operator"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn btn-operator"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn-operator"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn-operator"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="btn-zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn-operator"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
